Resources/Catalog/RucioFileCatalogClient.py

```diff
@@ -0,0 +1,590 @@
+            resDict['Successful'][lfn] = True
+          except Exception as err:
+            resDict['Failed'][lfn] = str(err)
+        else:
+          resDict['Failed'][lfn] = 'Not a file'
+      except DataIdentifierNotFound:
+        resDict['Failed'][lfn] = 'No such file or directory'
+      except Exception as err:
+        return S_ERROR(str(err))
+    return S_OK(resDict)
+
+
+  @checkCatalogArguments
+  def removeDirectory(self, lfns):
+    """ Remove the supplied directory
+    """
+    resDict = {'Successful': {}, 'Failed': {}}
+    for lfn in lfns:
+      try:
+        did = self.__getDidsFromLfn(lfn)
+        meta = self.client.get_metadata(did['scope'], did['name'])
+        if meta['did_type'] == 'DATASET':
+          try:
+            self.client.set_metadata(scope=did['scope'], name=did['name'], key='lifetime', value=1)
+            resDict['Successful'][lfn] = True
+          except Exception as err:
+            resDict['Failed'][lfn] = str(err)
+        else:
+          resDict['Failed'][lfn] = 'Not a directory'
+      except DataIdentifierNotFound:
+        resDict['Failed'][lfn] = 'No such file or directory'
+      except Exception as err:
+        return S_ERROR(str(err))
+    return S_OK(resDict)
+
+
+  @checkCatalogArguments
+  def getDirectorySize(self, lfns, longOutput=False, rawFiles=False):
+    """ Get the directory size
+    """
+    resDict = {'Successful': {}, 'Failed': {}}
+    for lfn in lfns:
+      try:
+        did = self.__getDidsFromLfn(lfn)
+        meta = self.client.get_metadata(did['scope'], did['name'])
+        if meta['did_type'] == 'FILE':
+          resDict['Failed'][lfn] = 'Not a directory'
+
+        elif meta['did_type'] == 'CONTAINER':
+          resDict['Successful'][lfn] = {'ClosedDirs': [], 'Files': 0, 'SiteUsage': {}, 'SubDirs': {}, 'TotalSize': 0}
+          for child in self.client.list_content(scope=did['scope'], name=did['name']):
+            childName = child['name']
+            if self.convertUnicode:
+              childName = str(childName)
+            resDict['Successful'][lfn]['SubDirs'][childName] = datetime.datetime.now()
+
+        else:
+          resDict['Successful'][lfn] = {'ClosedDirs': [], 'Files': 0, 'SiteUsage': {}, 'SubDirs': {}, 'TotalSize': 0}
+          for child in self.client.list_files(scope=did['scope'], name=did['name']):
+            resDict['Successful'][lfn]['Files'] += 1
+            resDict['Successful'][lfn]['TotalSize'] += child['bytes']
+          for rep in self.client.list_replicas([{'scope': did['scope'], 'name': did['name']}]):
+            fileSize = rep['bytes']
+            for rse in rep['rses']:
+              if rse not in resDict['Successful'][lfn]['SiteUsage']:
+                resDict['Successful'][lfn]['SiteUsage'][rse] = {'Files': 0, 'Size': 0}
+              resDict['Successful'][lfn]['SiteUsage'][rse]['Files'] += 1
+              resDict['Successful'][lfn]['SiteUsage'][rse]['Size'] += fileSize
```

Comment:
Can this be replaced by:
```suggestion
            resDict['Successful'][lfn]['SiteUsage'][rse] = {
                'Files': len(rep['rses']),
                'Size': len(rep['rses']) * rep['bytes'],
            }
```